plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt

```diff
@@ -48,40 +53,45 @@ internal data class Lockfile(
 }
 
 internal fun String.parseLockfile(): Lockfile {
-    val root = yamlMapper.readTree(this)
+    val root = Yaml.default.parseToYamlNode(this).yamlMap
+
+    val pods = root.yamlMap.get<YamlList>("PODS")!!.yamlList.items.map { node ->
+        when {
+            node is YamlMap -> {
+                val mapEntry = node.yamlMap.entries.entries.single()
+                val (name, version) = parseNameAndVersion(mapEntry.key.content)
 
-    val pods = root.get("PODS").map { node ->
-        when (node) {
-            is ObjectNode -> {
-                val (name, version) = parseNameAndVersion(node.fieldNames().asSequence().single())
                 Lockfile.Pod(
                     name = name,
                     version = version,
-                    dependencies = node.single().map {
-                        val (depName, depVersion) = parseNameAndVersion(it.textValue())
+                    dependencies = mapEntry.value.yamlList.items.map {
+                        val (depName, depVersion) = parseNameAndVersion(it.yamlScalar.content)
                         Lockfile.Pod(depName, depVersion)
                     }
                 )
             }
 
             else -> {
-                val (name, version) = parseNameAndVersion(node.textValue())
+                val (name, version) = parseNameAndVersion(node.yamlScalar.content)
                 Lockfile.Pod(name, version)
             }
         }
     }
 
-    val checkoutOptions = root.get("CHECKOUT OPTIONS")?.fields()?.asSequence()?.mapNotNull { (name, node) ->
+    val checkoutOptions = root.yamlMap.get<YamlMap>("CHECKOUT OPTIONS")?.entries.orEmpty().mapNotNull {
+        val name = it.key.content
+        val node = it.value.yamlMap
+
         val checkoutOption = CheckoutOption(
-            git = node[":git"]?.textValue(),
-            commit = node[":commit"].textValue()
+            git = node.get<YamlScalar>(":git")?.content.orEmpty(),
+            commit = node.get<YamlScalar>(":commit")?.content.orEmpty()
         )
 
         name to checkoutOption
-    }.orEmpty().toMap()
+    }.toMap()
 
-    val dependencies = root.get("DEPENDENCIES").map { node ->
-        val (name, version) = parseNameAndVersion(node.textValue())
+    val dependencies = root.yamlMap.get<YamlList>("DEPENDENCIES")?.items.orEmpty().map { node ->
```

Comment:
You already call `yamlMap` as part of defining `root`, so it can be dropped here.

Reply:
While I'm not a fan of using `!!`, the previous code assumed `get("DEPENDENCIES")` to always be non-null, so we probably use `!!` here to document that like you also did in line 58.